buying_selling/posts/api/views.py

```diff
@@ -23,39 +21,41 @@ def get_queryset(self, *args, **kwargs):
         return queryset
 
 
-class PostCreateAPIView(CreateAPIView):
+class PostViewset(viewsets.ModelViewSet):
+    """Manage posts in the database"""
+
+    serializer_action_classes = {
+        'create': PostCreateSerializer,
+        'list': PostListSerializer,
+        'retrieve': PostDetailSerializer,
+        'partial_update': PostUpdateSerializer,
+        'update': PostUpdateSerializer,
+    }
     queryset = Post.objects.all()
-    serializer_class = PostCreateSerializer
-    permission_classes = (IsAuthenticated,)
+
+    permission_classes_by_action = {
+        'create': [IsAuthenticated],
+        'list': [AllowAny],
+        'retrieve': [AllowAny],
+        'update': [IsAuthenticated, IsOwnerOrReadOnly],
+        'partial_update': [IsAuthenticated, IsOwnerOrReadOnly],
+        'destroy': [IsAuthenticated, IsOwnerOrReadOnly],
+    }
```

Comment:
@Saurav-Shrivastav Kindly verify the permission once to ensure I didn't make any mistakes

Reply:
Looks good to me.